### PR TITLE
Ia 924 install typescript

### PR DESCRIPTION
**Pull request by quang-le**
## Changes

- updating webpack files
- adding dependencies and scripts in package.json
- adding a `babel-register.js` files to prevent tests from crashing
- replacing `js-config.json`with `ts-config.json`
- updating `.eslintrc.json` to add TS linting and mirror js rules
- updating `.babelrc`
- updating webpack Dockerfile

Ticket with info for QA Testing: [IA-950](https://bluesquare.atlassian.net/browse/IA-950?atlOrigin=eyJpIjoiZTg5MmU5Zjc0ZmIyNDZhMWI4YTQwMzllNjQ1ZmUzOTIiLCJwIjoiaiJ9)

### CODE REVIEW

**Review by beygorghor — APPROVED** [review]
This is working for me

File: docker/webpack/Dockerfile [hunk 1]
```diff
@@ -11,9 +11,12 @@ RUN apt-get update && apt-get install -y \
 
 COPY package.json .
 COPY package-lock.json .
+COPY tsconfig.json .
+COPY babel-register.js .
+# COPY .babelrc .
```
Comment (olethanh) [inline docker/webpack/Dockerfile:16]: These files should be  part of the files we mount in the docker-compose so change are reflected directly without needing to rebuild the image

**Comment by olethanh** [comment]
I'm hitting this error in the eslintrc.json

```
Error: .eslintrc.json#overrides[0]:
	Configuration for rule "import/no-unresolved" is invalid:
	Value {"ignores":["modules"],"caseSensitive":true} should NOT have additional properties.


Error: .eslintrc.json#overrides[0]:
	Configuration for rule "import/no-unresolved" is invalid:
	Value {"ignores":["modules"],"caseSensitive":true} should NOT have additional properties.

    at ConfigValidator.validateRuleOptions (/home/olivier/Projects/blsq/iaso/node_modules/@eslint/eslintrc/lib/shared/config-validator.js:144:23)
    at /home/olivier/Projects/blsq/iaso/node_modules/@eslint/eslintrc/lib/shared/config-validator.js:199:18
    at Array.forEach (<anonymous>)
    at ConfigValidator.validateRules (/home/olivier/Projects/blsq/iaso/node_modules/@eslint/eslintrc/lib/shared/config-validator.js:196:34)
    at ConfigValidator.validateConfigArray (/home/olivier/Projects/blsq/iaso/node_modules/@eslint/eslintrc/lib/shared/config-validator.js:322:18)
    at CascadingConfigArrayFactory._finalizeConfigArray (/home/olivier/Projects/blsq/iaso/node_modules/@eslint/eslintrc/lib/cascading-config-array-factory.js:493:23)
    at CascadingConfigArrayFactory.getConfigArrayForFile (/home/olivier/Projects/blsq/iaso/node_modules/@eslint/eslintrc/lib/cascading-config-array-factory.js:299:21)
    at CLIEngine.isPathIgnored (/home/olivier/Projects/blsq/iaso/node_modules/eslint/lib/cli-engine/cli-engine.js:973:18)
    at ESLintPlugin.invokeESLint (/snap/pycharm-professional/265/plugins/JavaScriptLanguage/languageService/eslint/bin/eslint-plugin.js:89:23)
    at ESLintPlugin.getErrors (/snap/pycharm-professional/265/plugins/JavaScriptLanguage/languageService/eslint/bin/eslint-plugin.js:67:21)
Process finished with exit code -1
```

**Comment by olethanh** [comment]
I tried porting date.js file to Typescript. It works well. 
But I get a warning because it would like the extensions in the import `ESLint: Missing file extension "ts" for "../../utils/dates"(import/extensions)` not sure this is what we want? if it's not maybe we can deactivate this rule?

image

**Pull request merged**